File: win-hidemaru-app/hmAssemblyNGen.pl
# 秀丸のディレクトリに配置して実行すると
# アセンブリをNGenしてくれる

use File::Basename;
use FindBin;

my $selfdir = $FindBin::Bin;
my $exe='"C:\Windows\Microsoft.NET\Framework\v4.0.30319\ngen.exe"';

while (<"C:/usr/hidemaru/hm*/*.dll">) {
    my $dir = dirname($_);
    chdir dir;

    my $cmd = $exe . " install \"$_\" /AppBase:\"" . $selfdir . "/\"\n";
    print `$cmd`;
}

File: win-hidemaru-app/hm_assembly_ngen/hmAssemblyNGen.py
import os
import glob


thispath: str = os.path.dirname(os.path.abspath(__file__))

ngenpath: str= r"C:\Windows\Microsoft.NET\Framework\v4.0.30319\ngen.exe"

for targetfile in glob.glob(thispath + r"\hm*\*.dll"):
    targetdir: str = os.path.dirname(os.path.abspath(targetfile))
    os.chdir(targetdir)

    cmd: str = r'{} install "{}" /AppBase:"{}"'.format(ngenpath, targetfile, thispath)
    print(cmd)
    os.system(cmd)
